pandas/core/indexing.py

```diff
@@ -1615,6 +1615,12 @@ def _setitem_with_indexer(self, indexer, value):
                 info_idx = [info_idx]
             labels = item_labels[info_idx]
 
+            # Ensure we have something we can iterate over
+            ilocs = info_idx
+            if isinstance(info_idx, slice):
+                ri = Index(range(len(self.obj.columns)))
```

Comment:
use np.arange

Reply:
np.arange will give us an Int64Index, whereas this will give us a RangeIndex